optuna/samplers/_tpe/_truncnorm.py

```diff
@@ -151,14 +151,15 @@ def mass_case_central(a: np.ndarray, b: np.ndarray) -> np.ndarray:
 def _bisect(f: Callable[[float], float], a: float, b: float, c: float) -> float:
     if f(a) > c:
         a, b = b, a
-    # TODO(amylase): Justify this constant
-    for _ in range(100):
+    # In the algorithm, it is assumed that all of (a + b), (a * 2), and (b * 2) are finite.
+    while True:
```

Comment:
How about keeping the upper bound of the loop count? I am not confident that the condition `a == m or b == m` is satisfied in the convergent state in all environments.

Reply:
I see. I have revised the code.